apps/condo/domains/common/utils/serverSchema/changeTrackable.js

```diff
@@ -0,0 +1,355 @@
+ *
+ * Suppose, we have a `Ticket` schema, following workflow will happen:
+ * 1. A `Ticket` item is created
+ * 2. The item is updated
+ * 3. First `TicketChange` item is created
+ * 4. The `Ticket` item is updated second time
+ * 5. Second `TicketChange` item is created
+ *
+ * Each `TicketChange` item contains incremental changes of related `Ticket` ("source") schema item.
+ * A full snapshot is not stored by following reasons:
+ * 1. We already have system `…HistoryRecord` schema, added via `versioned` Keystone plugin;
+ * 2. Data bloating should be avoided.
+ *
+ *
+ * ## Integration guide
+ *
+ * For example, to track changes of `Ticket` schema, following should be done:
+ * 1. Decide, what fields should be trackable, using `trackableFieldsFrom` function;
+ * 2. Create a "destination" schema, e.g. `TicketChange`
+ * 3. Generate fields for it using `generateChangeTrackableFieldsFrom` function
+ * 4. Write `DisplayNameResolvers` (see description below in typedef)
+ * 5. Write `RelatedManyToManyResolvers` (see description below in typedef)
+ * 6. Implement an `afterChange` hook in `Ticket` and call `afterChangeHook` inside of it.
+ *
+ * That's it!
+ * Changes of `Ticket` will be stored in `TicketChange` schema!
+ *
+ * Remember to make migrations of `TicketChange` schema!
+ *
+ * ## Terms
+ *
+ * ### Display name
+ *
+ * String representation of related entity, supposed to be displayed in UI.
+ *
+ * ### Change storage set
+ *
+ * Suppose, we are going to store a change of a field, whose name is `…`.
+ * When the field is a scalar, following fields will be generated with the same type,
+ * as type of field in question:
+ *     - `…From`
+ *     - `…To`
+ * When the field is a `Relationship` with `many: true`, four fields will be generated:
+ *     - `…IdsFrom` – old list of ids of related entities;
+ *     - `…IdsTo` – new list of ids of related entities;
+ *     - `…DisplayNamesFrom` – old list of display names of related entities;
+ *     - `…DisplayNamesTo` – new list of display names of related entities.
+ * When the field is a single `Relationship`, four fields will be generated:
+ *     - `…IdFrom` – old list of ids of related entities;
+ *     - `…IdTo` – new list of ids of related entities;
+ *     - `…DisplayNameFrom` – old list of display names of related entities;
+ *     - `…DisplayNameTo` – new list of display names of related entities.
+ *
+ * @module changeTrackable
+ */
+
+/**
+ * Options for fields generator
+ * @typedef {Object} Options
+ * @property {string[]} except - Fields, that should not take part of change tracking
+ */
+
+/**
+ * Build set of fields to track, omitting those, specified in options.
+ *
+ * It's impossible to keep fields in constant somewhere, because we have a circular dependency:
+ *     - We need a schema to build fields from
+ *     - schema needs fields in it's hook.
+ *
+ * @param schema
+ * @param {Options} options
+ * @return {*}
+ */
+const trackableFieldsFrom = (schema, options = {}) => (
```

Comment:
It is better to minimize the number of functions available from outside the module.
You can move the use of the function inside `generateChangeTrackableFieldsFrom`

Reply:
Will be implemented in https://doma.atlassian.net/browse/SBERDOMA-391